developer/tutorials/articles/data-engine/01-intro.markdown

```diff
@@ -0,0 +1,114 @@
+# Data Engine
+
+Docs Outline
+
+1.  Introduction
+
+    Explain the overall goal: A Guestbook application where administrators can add
+    Guestbooks, while Guest Users can add Guestbook Entries to Guestbooks they're
+    interested in.
+
+    Administrators must have permission to manage (CRUD operations) Guestbooks,
+    which are the Data Record Collection.
+
+    The Guestbook Entry form is backed by the Data Engine Data Definition, where
+    Guest Users can add an Entry, so they must be able to view the Data Record
+    Collection and have CRUD operation ability over their data record.
+
+2.  Create the application project
+
+    We can provide a template here, or we can link to the React docs we
+    currently have, which basically say: 
+
+    Use `npm install -g yo generator-liferay-bundle`
+    Call `yo liferay-bundle`
+    Fill out CLI form accordingly
+
+3.  Create the Guestbook Entry Data Definition
+
+    Data structure: Name, Message text fields.
+
+    Validation: Both are required fields.
+
+    Permissions: Admins should be able to manage these with full CRUD operation support
+
+4.  Create the Guestbook Data Record Collection
+
+    Each Data Record collection represents a different Guestbook
+
+    Admins can do the full set of CRUD operations on these, from an
+    administrative UI (is that possible with data engine?)
+
+5.  Creating Data Layouts
+
+    Expose the administrative functions for the Guestbook Data Record
```

Comment:
'Expose the administrative functions for the Guestbook Data Record Collections'

This is nothing related to the layout. The layout is just how the data definition will be displayed to the user (i.e., whether the fields will be in the same column or not). The user is going to fill, and that information is going to be a Data Record to that particular Data Definition and Data Record Collection.